### ISSUE TITLE
Visualizing saliency for different labels

### ISSUE THREAD

**Issue by vjoshi48** [pytorch/captum#907]
I am trying to get the saliency visualization for my model which predicts 68 gray matter volumes. I want to get the saliency visualization for each of the 68 regions of interest to see if the model is correctly learning/using the image data to predict the volumes for regions of interest (eg, if the model is trying to predict the volume of the Amygdala, the saliency visualization should light up around the Amygdala).

The problem I am having currently, is that when I try and use the saliency.attribute function and set the target parameter to any number, I get the same exact visualization and same saliency tensor. I think I am misusing the library potentially and wanted to know why I can't get different saliency images for different labels. 

Here is a link to the code: https://github.com/vjoshi48/CatalystGriddingIssue
Here is a link to the PTH file needed to load model: https://drive.google.com/file/d/1G_Tm6yngrB8BNfFJH3u1kseZAjUxl6W-/view?usp=sharing

This is the line that I feel I am likely misunderstanding/misusing. I thought that if I set target=i (where i is the value of one of the 68 regions of interest for which I want the saliency), I would be able to get different visualizations.
attr_ig= saliency.attribute(
    features,
    target=i)

Let me know if you need any clarification/if you have any questions.

Thank you

**Comment by craymichael** [comment]
Apologies for the delay.

For a model output shaped `[batch, 68]`, passing `target=i` should select output dimension `i`. If every target gives exactly the same saliency tensor, I would check whether the model's forward pass is actually using separate output heads / weights for each region.

Two quick checks:

```python
out = model(features)
print(out.shape)  # should be [batch, 68]
print(out[:, 0], out[:, 1])
```

Then compare gradients manually:

```python
features.requires_grad_()
out = model(features)
g0 = torch.autograd.grad(out[:, 0].sum(), features, retain_graph=True)[0]
g1 = torch.autograd.grad(out[:, 1].sum(), features)[0]
print((g0 - g1).abs().max())
```

If `g0` and `g1` are identical, Captum is reflecting the model's gradients. If they differ manually but not through Captum, then the issue is likely in how the target or input batch is passed.

Also make sure `features` includes a batch dimension and that the model is in eval mode:

```python
model.eval()
```

This should answer the question, so I'll close this for now. Please feel free to reply with any follow-up questions.
